src/lib.rs

```diff
@@ -961,6 +961,47 @@ fn lexicographical_topological_sort(
     Ok(PyList::new(py, out_list).into())
 }
 
+#[pyfunction]
+fn graph_greedy_color(
+    py: Python,
+    graph: &graph::PyGraph,
+) -> PyResult<PyObject> {
+    let mut colors: HashMap<usize, usize> = HashMap::new();
+    let mut node_vec: Vec<NodeIndex> = graph.graph.node_indices().collect();
+    let node_count = graph.graph.node_count();
+    let mut sort_map: HashMap<NodeIndex, usize> = HashMap::new();
+    for k in node_vec.iter() {
+        // The intent behind this sort key is to sort nodes in descending order
+        // of degree
+        sort_map.insert(*k, node_count - graph.graph.edges(*k).count());
```

Comment:
comment is definitely helpful. How does `node_count` factor into this?

Reply:
`node_count` is just the number of nodes in the graph so we get a descending order by subtracting the degree from the max value. But now that you mention this there is a potential bug here with multigraphs I think. If the degree for 1 node is greater than the number of nodes this will overflow and have the opposite of the intended effect. This should be easy to fix, just change the value type in the sort_map from `usize` to `isize`. I'll push a fix and add a test

Reply:
So, my original idea of using isze would have too many tradeoffs for super large graphs (we lose too much between `usize::max` and `isize::max`). So, I tried the `Reverse` struct I mentioned eariler, by itself on `sort_by_key(|k| Reverse(graph.graph.edges(*k).count()))` I managed to get it working this time (I believe I made a typo the last time I tried it in my comments from a couple weeks ago) but this ended up being significantly slower the hash map approach @t-imamichi introduced. This is likely because `sort_by_key()` isn't caching the results from the key func, so it's calling `edges().count()` more than once for nodes. So I added back the hash map, but just store the degree for each node instead of subtracting and this gets us back to more or less parity with the numbers we had before:

```
--- fast-nx.txt	2020-06-22 14:07:58.020319946 -0400
+++ fast-numpy.txt	2020-06-22 14:08:04.167015153 -0400
@@ -1,4 +1,4 @@
-Elasped time: 5.515628322958946
+Elasped time: 6.124710976029746
```

Anyway, that's a long way of saying fixed in: https://github.com/Qiskit/retworkx/pull/77/commits/40c197fa71c2e14e3f67e5fa31adeb364d8cd667

Reply:
thanks 👍  :)